lib/publiccloud/utils.pm

```diff
@@ -839,22 +840,23 @@ sub wait_quit_zypper_pc {
     Optionally accepts proceed_on_failure => 1 to return undef instead of dying.
 
     Return:
-    - Array of CIDR strings, if found
-    - empty array otherwise
+    - worker ip, if retrieved
+    - undef otherwise (if proceed_on_failure is set)
 
 =cut
 
 sub detect_worker_ip {
     my (%args) = @_;
+    my $ip;
     for my $url ('http://checkip.amazonaws.com', 'https://ifconfig.me') {
-        my $ip = script_output("curl -q -fsS --max-time 10 $url",
+        $ip = script_output("curl -q -fsS --max-time 10 $url",
             timeout => 15, proceed_on_failure => 1);
         $ip =~ s/^\s+|\s+$//g;
-        next unless $ip;
+        next unless $ip && (inet_pton(AF_INET, $ip) || inet_pton(AF_INET6, $ip));
         return $ip;
     }
     return undef if $args{proceed_on_failure};
-    die "Worker IP could not be determined";
+    die "Worker IP could not be determined - return was $ip";
```

Comment:
yes, ok fine, but the printed value is potentially already mangled by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/23644/files#diff-27d3901d8dece7c398b9abdb53c98d17bad7e5c1cf9ed74f14ed9e68a6e6a1c2R854

`$ip =~ s/^\s+|\s+$//g;`

Reply:
True, but that just trims spaces, so no big harm done (I think)